src/sinks/amqp/sink.rs

```diff
@@ -54,6 +58,7 @@ pub(super) struct AmqpSink {
     pub(super) channel: Arc<lapin::Channel>,
     exchange: Template,
     routing_key: Option<Template>,
+    properties: Option<AmqpPropertiesConfig>,
```

Comment:
It might be slightly faster to make this a `BasicProperties` and do the build in `AmqpSink::new`. We then just `clone` for each event rather than `build`. 

I doubt if the difference will be much since we will still need to allocate for every event since `basic_publish` seems to want to take ownership of the properties for every call. So this isn't really that important.

Reply:
I considered the possibility of making some properties such as **correlation_id**, **client_id**, **cluster_id**, **app_id**, and **kind**, into **Template** type for future use. I prioritized extensibility.
Do I make the change anyway ?

Reply:
No. That's a valid point. This is good as it is.